Remotion/Web/Development.WebTesting/DownloadInfrastructure/Default/DefaultDownloadHelper.cs

```diff
@@ -183,6 +183,18 @@ private void CleanUpUnmatchedDownloadedFiles ([NotNull] IEnumerable<string> unma
       }
     }
 
+    private void WaitUntilTrueOrTimeout (Func<bool> func, TimeSpan timeout, TimeSpan interval)
+    {
+      var success = false;
+      var elapsed = 0d;
+      while (!success && elapsed < timeout.TotalMilliseconds)

```

Comment:
for-loop?

Reply:
Wouldn't a while loop be more expressive in this case, since we only care about not waiting too long and don't really care about how many times we try?

Reply:
yes, I wouldn't introduce a counter. just move the variables into the loop-head, the loop-condition into the for-condition and the elapsed increment into the loop-increment.

Reply:
for vs while: when there's dedicated start, exit, and increment statements, a for-loop is typically the correc tchoice because it introduces a wellknown structure. the while loop's typically chosen, when there's only an exit conditon or the increment can't extract to the loop-increment, e.g. because the increment happens because of the loops primary operation (e.g. stream reading)